server.js

```diff
@@ -27,6 +17,123 @@ app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+
+//GETS THE WHOLE DATA-SET UNFILTERED
+app.get('/collection', (req, res) => {
+  res.json(netflixData)
+})
+
+
+//GETS ALL MOVIES
+app.get('/movies', (req, res) => {
+  const onlyMovies = netflixData.filter((item) => item.type === "Movie")
+
+  res.json(onlyMovies)
+})
+
+
+//LOOKS FOR ALL MOVIES FROM A SPECIFIC YEAR AND ADDITIONAL QUERY FOR COUNTRY
+app.get('/movies/:year', (req, res) => {
+  const year = req.params.year
+  const onlyMovies = netflixData.filter((item) => item.type === "Movie")
+  let moviesFromYear = onlyMovies.filter((item) => item.release_year === +year)
+  const country = req.query.country
+
+  if (country) {
+    console.log(country)
+    moviesFromYear = moviesFromYear.filter((item) => item.country.toLowerCase() === country.toLowerCase())
+  }
+
+  res.json(moviesFromYear)
+})
+
+
+// !!THIS IS SUPPOST TO FILTER TYPE OF NETFLIX CONTENT AND COUNTRY WITH QUERIES BUT IT'S NOT WORKING
+//AND I DON'T UNDERSTAND WHY...
+
+// app.get('/collection', (req, res) => {
+//   let collection = netflixData
+//   const typeOfContent = req.query.type
+//   const showCountry = req.query.country
+
+//   if(typeOfContent) {
+//     collection = collection.filter((item) => item.type.toLowerCase() === typeOfContent.toLowerCase())
+//   }if(showCountry){
+//     collection = collection.filter((item) => item.country.toLowerCase() === showCountry.toLowerCase())
+//   }
+//   res.json(collection)
+// })
+
+
+//GETS ALL THE TV-SHOWS
+app.get('/tvshows', (req, res) => {
+  const onlyTvShows = netflixData.filter((item) => item.type === 'TV Show')
+  res.json(onlyTvShows)
+})
+
+
+//LOOKS FOR TV-SHOWS WITHIN A SPECIFIC GENRE
+app.get('/tvshows/:genre', (req, res) => {
+  const genre = req.params.genre.toString()
+  const onlyTvShows = netflixData.filter((item) => item.type === 'TV Show')
+  const specificGenreTv = onlyTvShows.filter((item) => item.listed_in.toLowerCase().includes(genre.toLowerCase()))
+
+  res.json(specificGenreTv)
+})
+
+
+//LOOKS FOR TV-SHOWS WITHIN A SPECIFIC GENRE AND FILTERS ON YEAR WITH QUERY
+//NOT WORKING!
+app.get('/tvshows:genre', (req, res) => {
+  const genre = req.params.genre
+  let onlyTvShows = netflixData.filter((item) => item.type === 'TV Show')
+  let specificGenreTv = onlyTvShows.filter((item) => item.listed_in.toLowerCase().includes(genre.toLowerCase()))
+  const year = req.query.year
+
+  if (year) {
+    specificGenreTv = specificGenreTv.filter((item) => item.release_year === +year)
+  }
+  res.json(specificGenreTv)
+})
```

Comment:
These routes are named the same as well. If you want to filter on multiple params, it's better to use query params. Here's a stack overflow question with an example: [https://stackoverflow.com/c/technigo/questions/1101](https://stackoverflow.com/c/technigo/questions/1101)